src/plugins/opencl_support/src/OpenCLSupportSettingsController.cpp

```diff
@@ -19,125 +19,150 @@
  * MA 02110-1301, USA.
  */
 
+#include "OpenCLSupportSettingsController.h"
+
+#include <QButtonGroup>
 #include <QLabel>
 #include <QLayout>
-#include <QCheckBox>
 
-#include <U2Core/AppContext.h>
 #include <U2Algorithm/OpenCLGpuRegistry.h>
-#include <U2Core/AppResources.h>
 
-#include "OpenCLSupportSettingsController.h"
+#include <U2Core/AppContext.h>
+#include <U2Core/AppResources.h>
 
 namespace U2 {
 
-OpenCLSupportSettingsPageController::OpenCLSupportSettingsPageController( const QString & _displayMsg, QObject * p /* = 0 */ ) :
-AppSettingsGUIPageController( tr("OpenCL"), OpenCLSupportSettingsPageId, p ), displayMsg(_displayMsg) {}
-
-AppSettingsGUIPageState * OpenCLSupportSettingsPageController::getSavedState() {
-    QList<OpenCLGpuModel *> registeredGpus = AppContext::getOpenCLGpuRegistry()->getRegisteredGpus();
-    OpenCLSupportSettingsPageState * s = new OpenCLSupportSettingsPageState( registeredGpus.size() );
-    for( int i = 0, end = s->enabledGpus.size(); i < end; ++i ) {
-        s->enabledGpus[i] = registeredGpus.at(i)->isEnabled();
-    }
+OpenCLSupportSettingsPageController::OpenCLSupportSettingsPageController(const QString &_displayMsg, QObject *p /* = 0 */)
+    : AppSettingsGUIPageController(tr("OpenCL"), OpenCLSupportSettingsPageId, p), displayMsg(_displayMsg) {
+}
 
-    return s;
+AppSettingsGUIPageState *OpenCLSupportSettingsPageController::getSavedState() {
+    return new OpenCLSupportSettingsPageState(AppContext::getOpenCLGpuRegistry()->getEnabledGpuName());
 }
 
-void OpenCLSupportSettingsPageController::saveState( AppSettingsGUIPageState * _s ) {
+void OpenCLSupportSettingsPageController::saveState(AppSettingsGUIPageState *_s) {
     QList<OpenCLGpuModel *> registeredGpus = AppContext::getOpenCLGpuRegistry()->getRegisteredGpus();
-    OpenCLSupportSettingsPageState * s = qobject_cast<OpenCLSupportSettingsPageState *>(_s);
+    OpenCLSupportSettingsPageState *s = qobject_cast<OpenCLSupportSettingsPageState *>(_s);
 
     //saving state of enabled/disabled GPUs into registry
-    for( int i = 0, end = s->enabledGpus.size(); i < end; ++i ) {
-        registeredGpus[i]->setEnabled( s->enabledGpus[i] );
+    const QString enabledGpu = s->getEnabledGpuName();
+    bool enabledGpuWasFound = false;
+    foreach (OpenCLGpuModel *gpu, registeredGpus) {
+        if (enabledGpu.startsWith(gpu->getName())) {
```

Comment:
They should be equal, shouldn't they?

Reply:
Since I've added QMap with the name of GPU as keys - yes, but, earlier, the "enabledGpu" was a sign near the QRadioButton and this sign also includes the information about the memory.

Reply:
But you've already added a QMap? So, it is time to replace it to equality sign.